pkg/controller/operands/kubevirt.go

```diff
@@ -294,35 +295,36 @@ func (h *kvConfigHooks) addEnabledFeatureGates(req *common.HcoRequest, foundFgSp
 
 func filterOutDisabledFeatureGates(req *common.HcoRequest, foundFgSplit []string, resultFg []string) (bool, []string) {
 	fgChanged := false
+	featureGateChecks := getFeatureGateChecks(req.Instance.Spec.FeatureGates)
 	for _, fg := range foundFgSplit {
-		// Remove if not in HC CR
-		switch fg {
-		case HotplugVolumesGate:
-			if !req.Instance.Spec.FeatureGates.IsHotplugVolumesEnabled() {
-				fgChanged = true
-				continue
-			}
-		case kvWithHostPassthroughCPU:
-			if !req.Instance.Spec.FeatureGates.IsWithHostPassthroughCPUEnabled() {
-				fgChanged = true
-				continue
-			}
-		case kvWithHostModelCPU:
-			if !req.Instance.Spec.FeatureGates.IsWithHostModelCPUEnabled() {
-				fgChanged = true
-				continue
-			}
-		case SRIOVLiveMigrationGate:
-			if !req.Instance.Spec.FeatureGates.IsSRIOVLiveMigrationEnabled() {
-				fgChanged = true
-				continue
-			}
+		if isFeatureGateMissingFrom(featureGateChecks, fg) {
+			fgChanged = true
+			continue
 		}
 		resultFg = append(resultFg, fg)
 	}
 	return fgChanged, resultFg
 }
 
+func isFeatureGateMissingFrom(checks featureGateChecks, featureGate string) bool {
+	if check, isKnown := checks[featureGate]; isKnown {
+		return !check()
+	}
+	return false
```

Comment:
I think it should be `true` here. If we get here, the FG **is** missing. Do you have a test to cover this (It's a bit confusing).

Reply:
I agree, switched it over while testing.
Will add a test.